engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoDateTime.java

```diff
@@ -205,26 +202,6 @@ ZoneId asTimeZone() {
     return dateTime.getZone();
   }
 
-  @ExportMessage
-  Type getMetaObject(@CachedLibrary("this") InteropLibrary thisLib) {
-    return EnsoContext.get(thisLib).getBuiltins().dateTime();
-  }
-
-  @ExportMessage
-  boolean hasMetaObject() {
-    return true;
-  }
-
-  @ExportMessage
-  boolean hasType() {
-    return true;
-  }
-
-  @ExportMessage
-  Type getType(@Bind("$node") Node node) {
-    return EnsoContext.get(node).getBuiltins().dateTime();
```

Comment:
Previously the `EnsoContext.get` contained the dynamic lookup of _context_. The next call `getBuiltins().dataTime()` was a compile time constant derived from the _context_ (which was also compile time contexts thanks to #7493).

Reply:
The latest revision 7969907da39bcb886220b8a3c102f1943c08cd52 does not use cached `WeakReference` to cache the context anymore.